src/util/fetch.ts

```diff
@@ -0,0 +1,18 @@
+import {getCookie , setCookie, deleteCookie} from '../util/Authentication'
+
+export const fetchAPIData = async (url: string, method: string, body:object , headers?:object) => {
+    const cookie = getCookie('shelf-share-session')as { token: string }
+    const token = cookie.token
```

Comment:
Can you make this cookie?.token instead (the question mark is called optional chaining in case you are unfamiliar)? When cookie is undefined this causes an error.